magenta-lib/src/main/scala/magenta/tasks/AWS.scala

```diff
@@ -81,6 +82,24 @@ object S3 extends AWS {
     }
     bucketName
   }
+
+  def resolveBucket(bucket: Bucket, ssmClient: => SsmClient, reporter: => DeployReporter): String = {
```

Comment:
we have two `resolveBucket` functions, the one above turn the deploy data into a `Bucket` wheras this one turns that `Bucket` into a String.
I'd prefer them to have different names (and resolve is a bit of a generic word) - could we call the top one something like `getMagentaBucket` and this one `getAWSBucketName` or something?